lib/coverband/utils/file_path_helper.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+####
+# Helper functions for shared logic related to file path manipulation
+####
+module Coverband
+  module Utils
+    module FilePathHelper
+      ###
+      # Takes a full path and converts to a relative path
+      ###
+      def full_path_to_relative(full_path)
+        relative_filename = full_path
+        root_paths.each do |root|
+          relative_filename = relative_filename.gsub(/^#{root}/, './')
+          # once we have a relative path break out of the loop
+          break if relative_filename.start_with? './'
+        end
+        relative_filename
+      end
+
+      ###
+      # relative_path_to_full code takes:
+      # relative_path: which is a full path the same as reported by Coverage
+      # roots: if a collection of all possible full app paths
+      #    EX: [Coverband.configuration.root_paths, "#{current_root}/"]
+      # The LAST item should be the current file system root
+      # it expands that expands and adds a '/' as that isn't there from Dir.pwd
+      #
+      # NOTEs on configuration.root_paths usage
+      # strings: matching is pretty simple for full string paths
+      # regex: to get regex to work for changing deploy directories
+      #        the regex must be double escaped in double quotes
+      #          (if using \d for example)
+      #        or use single qoutes
+      #        example: '/box/apps/app_name/releases/\d+/'
+      #        example: '/var/local/company/company.d/[0-9]*/'
+      ###
+      def relative_path_to_full(relative_path, roots)
+        relative_filename = relative_path
+        local_filename = relative_filename
+        roots.each do |root|
+          relative_filename = relative_filename.gsub(/^#{root}/, './')
```

Comment:
If this file is relative, why do we need to chop off the root?

Reply:
Not true for Gem data, gems have gem_home which is one of the roots